src/config.rs

```diff
@@ -138,6 +221,12 @@ impl Config {
                 options.read(true);
                 options.write(true);
 
+                if !self.use_os_cache && cfg!(unix) {
+                    use std::os::unix::fs::OpenOptionsExt;
+                    options.custom_flags(libc::O_DIRECT);
+                    panic!("O_DIRECT support not sussed out yet.");
```

Comment:
this will eventually let us completely bypass the OS cache. when using compression, the OS cache just stores compressed disk blocks, which is less useful to us than the decompressed part, so sometimes you want to disable it.